src/CMakeLists.txt

```diff
@@ -54,6 +54,14 @@ set(DEP_LIBS_PKG_ARG_LISTS "IpcShm ${DEP_IPC_SHM_VERSION} CONFIG")
 # (TODO: but look into the jemalloc/jemalloc-cmake GitHub repo -- what's that all about?)
 # Hence we must ensure 2 things more manually: the lib location and the include-path.
 
+# Try pkg-config if available
```

Comment:
There are a few related things going on possibly outside the issues you were trying to (and succeeding to) fix/improve, so I might make a new PR after looking into everything. I omit details here to avoid extra confusion. Anyway I shall look into that soon but probably not right this second.

Even before then, let me put some comments/questions here, before I forget. Your answers (if any) will serve to at least educate me. (Undoubtedly I am a CMake noob; in fact I learned it, to the extent that I did, for this project.)

--

- This looks delightful -- but shouldn't this be within `if(NOT JEMALLOC_LIB)`? To the extent that this is better than find_library() and all that, perhaps it should be attempted a bit below, and if it fails then go on to my find_library() stuff?
- I am also curious; and again this is most likely just for my own education; what moved you to make this change? Was it a general desire to improve this, or did you observe (or know of a use case where one would observe) the find_library() stuff failing, where pkg-config would have succeeded? If it's only a general desire to improve this, that doesn't mean we shouldn't; I just want to know the reason is all.

Reply:
> Shouldn't this be within `if(NOT JEMALLOC_LIB)`?

Agree. This shoubld be within `if(NOT JEMALLOC_LIB)`.

Personally I perfer the following logic:

1. Use user specified elements (but via other variable names, explanation later).
2. If available, use `find_package()` to find corresponding elements.
3. Else, use `find_package(PkgConfig)` and the following `pkg_check_modules()` (which drives `pkgconf` or old `pkg-config`) to find corresponding elements.
4. (Not recommended) manually call `find_library()` and `find_path()` and set corresponding elements.

Usually `find_package()` is the approach recommended by CMake, but regrettably `jemalloc` does not provided files used by `find_package()`. `jemalloc.pc` is provided and much more cleaner than using `find_library()` and `find_path()` directly.